File: Examples/Getting-Started/Get-Device-Interfaces/nodeJS/get-device-interfaces.mjs
import "dotenv/config";

import { NetLDClient, printDeviceInterfaces } from "./netld-example-client.mjs";

const client = NetLDClient.fromEnv();
await client.login();

const interfaces = await client.getDeviceInterfaces({
  network: process.env.NETLD_NETWORK || "Default",
  ipAddress: process.env.NETLD_DEVICE_IP || "10.95.1.40",
});

printDeviceInterfaces(interfaces);

File: Examples/Getting-Started/Get-Device-Interfaces/Python/get_device_interfaces.py
import os

from netld_example_client import NetLDClient, print_device_interfaces


client = NetLDClient.from_env()
client.login()

interfaces = client.get_device_interfaces(
    network=os.environ.get("NETLD_NETWORK", "Default"),
    ip_address=os.environ.get("NETLD_DEVICE_IP", "10.95.1.40"),
)

print_device_interfaces(interfaces)
